Jira/nms 15584 (#6033)
OpenNMS/opennms: docs/modules/operation/pages/deep-dive/alarms/alarm-handling.adoc

== Alarm handling

The following interactions with alarms are possible.

=== Acknowledgment

PICTURES MISSING

> An alarm can be acknowledged by an user.

Reviewer: ```suggestion
Users can acknowledge alarms.
```

> The alarm will be moved out from `Alarm(s) outstanding` into `Alarm(s) acknowledged` view.

Reviewer: ```suggestion
The alarm will be moved out from `Alarm(s) outstanding` into the `Alarm(s) acknowledged` view.
```

By doing this other OpenNMS users can notice, that someone is aware of this alarm.

> NOTE: A manually acknowledged alarm that was cleared automatically by its clear event needs also unacknowledged manually to disappear.

Reviewer: I'm not clear what you mean here. 
Alarm was manually acknowledged. Clear event cleared it. They then need to manually unacknowledge it? Is that because it remains visible as long as it is marked acknowledged?

Author: > Alarm was manually acknowledged. Clear event cleared it. They then need to manually unacknowledge it?

Yes. That is the reality. You need to unack it to let it disappear from the alarm list.

> Is that because it remains visible as long as it is marked acknowledged?

That is also a good explanations. Acked alerts remain visible even if a clear event cleared it.

Reviewer 2: This isn't true.  _All_ alarms will eventually be removed automatically.  It's just that ack'ed alarms are kept longer than un-acked alarms.

Reviewer 2: Per `alarmd.drl`, un-acked alarms are kept for 5 minutes after clearing.  Acked alarms are kept for 1 day after clearing.  This is customizable by the user editing the cleanup rules in the alarmd drools.

Author: I am pretty sure I saw this behavior often in my system. And I didn't change drools files for sure.
But let me add @mmahacek 's point into the docs, since they reflect how it should work. When I see it in the real world again, I can open a bug report ;-)

Reviewer 2: Default alarmd cleanup interval below.  This is based on the last event time, so any new event coming in and getting reduced into an existing alarm will reset this timer.  We try to purge alarms from the database as any open alarm requires overhead in drools to keep track of it.  Not much of an issue on smaller instances, but can become more resource intensive when there's a high number of active alarms.

|Criteria|Cleanup Interval|
|---|---|
|Cleared and Un-ack|5 mins|
|Cleared and Acked|1 day|
|All Un-ack|3 days|
|All alarms|8 days|

=== Clearing

PICTURES MISSING

Clearing an oustanding or acknowledged alarm means "mark it as resolved".

> The cleared alarm will disappear within a defined cycle.

Reviewer: But above we have the note about having to still unaknowledge it.

NOTE: If the alarm was created by a threshold configuration the alarm will appear again, if the defined metric is still or again matching the threshold configurations.
An alarm based on a `nodeLostService` event won't appear again. To keep track on those `outage alarms` check the outages list.

> === Escalation

Reviewer: ```suggestion
== Escalation
```

PICTURES MISSING

> An alarm always has the same <<deep-dive/events/event-configuration.adoc#severities, severities>> like its underlaying event.

Reviewer: ```suggestion
An alarm always has the same <<deep-dive/events/event-configuration.adoc#severities, severities>> as its underlaying event.
```

If an alarm gets escalated, the alarm's <<deep-dive/events/event-configuration.adoc#severities, severity>> gets increased one level.

> === Notes

Reviewer: ```suggestion
== Alarm notes
```

{page-component-title} creates alarms for issues in the network.
Alarm notes let you assign comments to a specific alarm, or to a whole class of alarms, and share that information with other people on your team.

.Alarm Details page with sample notes
image::alarms/01_alarm-notes.png["Alarm Details page displaying sample notes in the Sticky Memo and Journal Memo boxes", 850]

You can add two types of notes to existing alarms or alarm classes:

* *Sticky Memo:* A user-defined note for a specific instance of an alarm.
Deleting the alarm also deletes any associated sticky memos.
* *Journal Memo:* A user-defined note for a class of alarms, based on the resolved reduction key.
Journal memos are shown for all alarms that match a specific reduction key.
Deleting an individual alarm does not remove the journal memo.
+
You must click *Clear* on an alarm with an associated journal memo to remove it.

> Symbols on the Alarm List Summary and Alarm List Detail pages indicate if individual alarms have sticky or journal memos associated.

Reviewer: ```suggestion
Symbols on the Alarm List Summary and Alarm List Detail pages indicate if individual alarms have associated sticky or journal memos.
```

> = Alarm handling

Reviewer 2: ```suggestion
= Alarm Handling
```

The following are ways you can interact with alarms.

== Acknowledgment

PICTURES MISSING

Users can acknowledge alarms.
The alarm will be moved out from `Alarm(s) outstanding` into the `Alarm(s) acknowledged` view.
Doing this lets other {page-component-title} users see that someone is aware of the alarm.

> 
> NOTE: A manually acknowledged alarm that was cleared automatically by its clear event needs also unacknowledged manually to disappear.

Reviewer 2: ```suggestion
```

== Clearing

PICTURES MISSING

Clearing an outstanding or acknowledged alarm means "mark it as resolved".
The cleared alarm will disappear within a defined cycle.

> NOTE: Alarms created by a threshold configuration will appear again if the defined metric still matches the threshold configurations or matches it again.
> An alarm based on a `nodeLostService` event won't appear again.
> To keep track of those `outage alarms`, check the outages list.

Reviewer 2: Any new event with alarm data will re-open a cleared alarm. Technically a new nodeLostService event will re-open an existing cleared alarm.  It's just that the system won't generate another copy of that event unless the service comes back online.
```suggestion
```

Author: Maybe the difference here is, how the alarm was cleared or if there is still an outage entry for it. When I clear a `nodeLostService` alarm manually, it gets removed after 5 mins. I still have an outage for this service. Then I will never get a new alarm for this one anymore. Also a restart of ONMS won't trigger this.
But this is different for threshold alarms. If new collection samples get checked and the threshold fires, I will get an alarm again.

This is what I wanted to tell here.

Reviewer 2: it is more a matter if the event will be re-sent.  Once a service is down, there won't be another down event generated unless the event comes back up.  Threshold events can be re-sent if the threshold is detected as breached again.

== Escalation

PICTURES MISSING

An alarm always has the same <<deep-dive/events/event-configuration.adoc#severities, severities>> as its underlaying event.

> If an alarm gets escalated, the alarm's <<deep-dive/events/event-configuration.adoc#severities, severity>> increases one level.

Reviewer 2: Don't need to link to the same destination two sentences in a row.
```suggestion
If an alarm gets escalated, the alarm's severity increases one level.
```

== Alarm notes

> {page-component-title} creates alarms for issues in the network.

Reviewer 2: ```suggestion
```

The following are ways you can interact with alarms.

== Acknowledgment

Users can acknowledge alarms to let other {page-component-title} users see that someone is aware of the alarm.
The alarm will be moved out from `Alarm(s) outstanding` into the `Alarm(s) acknowledged` view.
Acknowledged alarms will also be hidden from the "Nodes with Pending Problems" section of the home page.

.Acknowledged alarm of an HTTP outage in the alarm overview
image::alarms/acked_alarm_overview.png["Acknowledged alarm of an HTTP outage in the alarm overview"]

.Acknowledged alarm of an HTTP outage in detail view
image::alarms/acked_alarm_detail.png["Acknowledged alarm of an HTTP outage in detail view"]

== Clearing

Clearing an alarm means "mark it as resolved".

.Cleared alarm of an HTTP outage in the alarm overview
image::alarms/cleared_alarm_overview.png["Cleared alarm of an HTTP outage in the alarm overview"]

.Cleared alarm of an HTTP outage in detail view
image::alarms/cleared_alarm_detail.png["Cleared alarm of an HTTP outage in detail view"]

== Escalation

By default, an alarm has the same <<deep-dive/events/event-configuration.adoc#severities, severity>> as its most recent event.

> If an alarm gets escalated, the alarm's severity is increased by one level.

Reviewer: ```suggestion
If an alarm gets escalated, the alarm's severity increases by one level.
```

.Escalated alarm of an HTTP outage in the alarm overview
image::alarms/escalated_alarm_overview.png["Escalated alarm of an HTTP outage in the alarm overview"]

.Escalated alarm of an HTTP outage in detail view
image::alarms/escalated_alarm_detail.png["Escalated alarm of an HTTP outage in detail view"]

== Related events

> {page-component-title} correlates possible related events into alarms based on events with the same <<deep-dive/alarms/configuring-alarms#ga-reduction-key, reduction key>> to do this.

Reviewer: ```suggestion
{page-component-title} correlates possible related events into alarms based on events with the same <<deep-dive/alarms/configuring-alarms#ga-reduction-key, reduction key>>.
```
